pycbc/workflow/segment.py

```diff
@@ -90,23 +93,27 @@ def get_segments_file(workflow, name, option_name, out_dir):
 
     # Check for veto definer file
     veto_definer = None
-    if cp.has_option("workflow-segments", "segments-veto-definer-url"):
-        veto_definer = save_veto_definer(workflow.cp, out_dir, [])
+    veto_tags = []
+    if 'veto' in workflow.cp.get_subsections('workflow-segments'):
+        veto_tags = ['veto']
+    if cp.has_option_tags("workflow-segments", "segments-veto-definer-url", veto_tags):
+        veto_definer = save_veto_definer(workflow.cp, out_dir, veto_tags)
```

Comment:
I think this function should take the tag, but I don't think this should be hardcoded. I think it should just take `tags` sent to this function.

Reply:
I am worried that this would duplicate the veto definer being saved, but maybe the better move is change the veto-definer to be in `workflow-segments-veto` in the config files??

Reply:
The hardcoded veto tag is necessary. This method gets called twice, once with the passed in tag being 'veto' and once with it being 'science' . The 'science' tagged job still needs access to the veto definer which means it needs to look in `workflow-segments-veto` for the options relevant to the veto definer. 

I am slightly confused about the 2nd comment, the veto definer options ARE in `workflow-segments-veto` as shown here: https://git.ligo.org/ligo-cbc/pycbc-config/-/merge_requests/376/diffs

Reply:
Thanks for the changes. This is the only point that I think still needs discussion. The real concern is that `get_segments_file` should not be hardcoded around requirements of the all-sky workflow, and I think the design decision should be "how should this function work" and then "how do we call it", and not "how do we make this work for the way that it's called in the all-sky workflow".

I definitely think it's confusing that if I was (for some reason) using two veto-definer files, I wouldn't get the one I expected.

I also note that the `save_veto_definer` function doesn't actually work right with tags. It uses the tags to *identify* the veto-definer URL, but then writes the location back to `[workflow-segments]` ignoring if a tag was used. That would also be bad.

So, to avoid expanding the scope of this PR to fixing broken parts of functions, but still allowing us to solve the problem you had I would suggest:

* Please remove the hardcoded `veto` tag in `get_segments_file`. It's fine to hardcoded tags in the top-level workflow generators as those are specifying a complete workflow, they should not be hardcoded in utility functions like this one though.
* Do not send *any* tags through to `save_veto_definer` (because it doesn't work).

(So basically go back to what was previously done for the call to `save_veto_definer`). This does then require that there is only one veto-definer, and it is given in the top-level `[worflow-segments]`. If someone wanted to use multiple veto-definer files we could consider how to make this work properly (which might involve sending the veto-definer as a File object to this function, if needed.

Does that work?

Also note that I'm not looking at any changes in LVK-proprietary repositories when reviewing changes in the public PyCBC repository. (My comment was moot anyway when I realised that `save_veto_definer` doesn't really work as intended with tags). We should consider changes to the [not-LVK] example config files here:

https://github.com/gwastro/pycbc/tree/master/examples/search

but as veto-definers are not public (GWOSC handles this differently, but provides the same information), it's not relevant in this case.

Reply:
ok that all makes sense. I don't see us using multiple veto definer files anytime soon (what would be the point? the veto definer already contains multiple tables, one per ifo).